tests/integration/test_nft_creation.py

```diff
@@ -0,0 +1,418 @@
+    get_publisher_wallet,
+)
+from web3 import exceptions
+from ocean_lib.web3_internal.wallet import Wallet
+from web3 import Web3
+
+
+def deploy_erc721_erc20(web3, config, erc721_publisher: Wallet, erc20_minter: Wallet):
+    """Helper function to deploy an ERC721Token using erc721_publisher Wallet
+    and an ERC20Token data token with the newly ERC721Token using erc20_minter Wallet.
+    """
+
+    erc721_factory = ERC721FactoryContract(web3, get_nft_factory_address(config))
+    tx = erc721_factory.deploy_erc721_contract(
+        "NFT",
+        "NFTSYMBOL",
+        1,
+        ZERO_ADDRESS,
+        "https://oceanprotocol.com/nft/",
+        erc721_publisher,
+    )
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    registered_event = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_NFT_CREATED,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    token_address = registered_event[0].args.newTokenAddress
+    erc721_token = ERC721Token(web3, token_address)
+
+    erc_create_data = ErcCreateData(
+        1,
+        ["ERC20DT1", "ERC20DT1Symbol"],
+        [
+            erc20_minter.address,
+            erc721_publisher.address,
+            erc721_publisher.address,
+            ZERO_ADDRESS,
+        ],
+        [web3.toWei("0.5", "ether"), 0],
+        [b""],
+    )
+    tx_result = erc721_token.create_erc20(erc_create_data, erc721_publisher)
+    tx_receipt2 = web3.eth.wait_for_transaction_receipt(tx_result)
+
+    registered_event2 = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_TOKEN_CREATED,
+        tx_receipt2.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+
+    erc20_address = registered_event2[0].args.newTokenAddress
+
+    erc20_token = ERC20Token(web3, erc20_address)
+    return erc721_token, erc20_token
+
+
+def test_erc721_roles(web3, config):
+    """Test erc721 implicit and explicit role assignments  as well as removing them"""
+
+    # NFT Owner is also added as manager when deploying (first time), if transferred that doesn't apply
+    publisher = get_publisher_wallet()
+    consumer = get_consumer_wallet()
+    another_consumer = get_another_consumer_wallet()
+
+    erc721_factory = ERC721FactoryContract(web3, get_nft_factory_address(config))
+    tx = erc721_factory.deploy_erc721_contract(
+        "NFT", "NFTSYMBOL", 1, ZERO_ADDRESS, "https://oceanprotocol.com/nft/", publisher
+    )
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    registered_event = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_NFT_CREATED,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    assert registered_event[0].event == "NFTCreated"
+    assert registered_event[0].args.admin == publisher.address
+    token_address = registered_event[0].args.newTokenAddress
+    erc721_token = ERC721Token(web3, token_address)
+
+    """ 
+    permissions
+    0: manager
```

Comment:
we should have these in some sort of convention, maybe as part of agreements. Do not use numbers like this, directly. Let's use something like ERC721Permissions.MANAGER etc.